app/controllers/items_controller.rb

```diff
@@ -1,7 +1,29 @@
 class ItemsController < ApplicationController
 
   def browse
-    @res = $api.query({"facet_num" => 500, "facet_sort" => "term|asc" }).facets
+  end
+
+  def browse_facet
+    # Reverse facet name from url-formatting
+    @browse_facet = params[:facet]
+    if @browse_facet.include?("-")
```

Comment:
The URLs created for `date.year` and `creator.name` with `datify_name` are `/browse/date-year` and `/browse/creator-name`, so if the facet name parameter from the URL includes a dash, I apply the `datify_name` method on each facet's key and compare. If they match, I know that's the facet whose link was clicked.

I think we planned to allow `-` in field names, so I think this will handle any possible mix of facet names containing `-` and `.`, rather than assuming all `-`s in a key from the URL were originally `.`s.

The only mixup I can see is if we had some strange situation where both `abc-abc.abc` and `abc.abc.abc` were both facets. I can't see where that'd happen atm though.